PWGHF/HFL/HFmAnalysis.cxx

```diff
@@ -0,0 +1,339 @@
+// Copyright 2019-2020 CERN and copyright holders of ALICE O2.
+// See https://alice-o2.web.cern.ch/copyright for details of the copyright holders.
+// All rights not expressly granted are reserved.
+//
+// This software is distributed under the terms of the GNU General Public
+// License v3 (GPL Version 3), copied verbatim in the file "COPYING".
+//
+// In applying this license CERN does not waive the privileges and immunities
+// granted to it by virtue of its status as an Intergovernmental Organization
+// or submit itself to any jurisdiction.
+
+/// \file HFmAnalysis.cxx
+/// \brief Task derived from the DQ framework and used to extract the observables on single muons needed for the HF-muon analysis.
+/// \author Maolin Zhang <maolin.zhang@cern.ch>, CCNU
+
+#include "Framework/runDataProcessing.h"
+#include "Framework/AnalysisTask.h"
+#include "Framework/ASoAHelpers.h"
+#include "Framework/AnalysisDataModel.h"
+#include "PWGDQ/Core/VarManager.h"
+#include "PWGDQ/Core/HistogramManager.h"
+#include "PWGDQ/Core/AnalysisCut.h"
+#include "PWGDQ/Core/AnalysisCompositeCut.h"
+#include "PWGDQ/Core/CutsLibrary.h"
+#include "PWGDQ/Core/HistogramsLibrary.h"
+#include "Framework/HistogramRegistry.h"
+#include "PWGDQ/DataModel/ReducedInfoTables.h"
+#include "ReconstructionDataFormats/TrackFwd.h"
+#include "Common/DataModel/EventSelection.h"
+
+using namespace o2;
+using namespace o2::aod;
+using namespace o2::framework;
+
+namespace o2::aod
+{
+namespace dq_analysis_flags
+{
+DECLARE_SOA_COLUMN(IsEventSelected, isEventSelected, int);
+}
+
+DECLARE_SOA_TABLE(EventCuts, "AOD", "EVENTCUTS", dq_analysis_flags::IsEventSelected);
+} // namespace o2::aod
+
+using MyCollisions = o2::soa::Join<aod::Collisions, aod::EvSels>;
+using MyMcCollisions = o2::soa::Join<aod::Collisions, aod::McCollisionLabels, aod::EvSels>;
+using MyEventsSelected = o2::soa::Join<aod::Collisions, aod::EvSels, o2::aod::EventCuts>;
+using MyMcEventsSelected = o2::soa::Join<aod::Collisions, aod::McCollisionLabels, aod::EvSels, o2::aod::EventCuts>;
+using MyMuons = o2::soa::Join<aod::FullFwdTracks, aod::McFwdTrackLabels>;
+
+constexpr static uint32_t gMuonFillMap(VarManager::ObjTypes::Muon | VarManager::ObjTypes::MuonCov);
+constexpr static uint32_t gEventFillMap(VarManager::ObjTypes::BC | VarManager::ObjTypes::Collision);
+constexpr static uint32_t gTrackMCFillMap(VarManager::ObjTypes::ParticleMC);
+
+void defineHistograms(HistogramManager*, TString);
+
+struct HfmEventSelection {
+  Configurable<bool> applySoftwareTrigger{"appllySoftwareTrigger", false, "whether to apply the software trigger"};
+  Configurable<int> softwareTrigger{"softwareTrigger", VarManager::kIsMuonSingleLowPt7, "software tigger flag"};
```

Comment:
Still the same typo.
```suggestion
  Configurable<int> softwareTrigger{"softwareTrigger", VarManager::kIsMuonSingleLowPt7, "software trigger flag"};
```

Reply:
Thanks for you suggestions and I do need to apology for these continuous commits. I will learn how to make it more efficient.
I am trying to find the variables with bad names and I will fix them, but there are some variables with bad names (like fHistMan, fValues) I found working well in PWGDQ codes, do I need to fix them too?
As for the directory structure. Is it okay to create a Tasks folders under the HFL folder and move the code there ? O2Physics/PWGHF/HFL/Tasks/
As I understand from the presentation from last year the validation framework is for testing and validation (w.r.t Run2).
 https://indico.cern.ch/event/1101005/contributions/4632407/attachments/2360172/4028725/2021-12-07_HF-O2-hackathon_O2.pdf
 Is it required for running this code on the AO2Ds ? Incase I misunderstood it, please let me know. 

 Thanks a lot for your help.

Reply:
- I don't know if in PWG-DQ they have some conventions for the meaning of the `f` prefix, but to me it doesn't have any obvious purpose. It looks more like imported from an AliPhysics class but I might be wrong.
- Putting the code in the `HFL/Task` directory is fine unless it is meant to produce tables processed by another task.
- The validation framework is a convenient tool to run the HF analyses locally (as described in the PWG-HF documentation). It is recommended to add a new task there so that anyone can easily run it.